lib/ecto_ip_range/postgrex/ip_range_extension.ex

```diff
@@ -0,0 +1,124 @@
+defmodule EctoIPRange.Postgrex.IPRangeExtension do
+  @moduledoc """
+  Postgrex extension for "iprange" fields.
+  """
+
+  use Postgrex.BinaryExtension, type: "iprange"
+
+  import Postgrex.BinaryUtils, warn: false
+
+  alias EctoIPRange.{
+    IPRange,
+    Util.CIDR,
+    Util.Inet
+  }
+
+  def encode(_) do
+    quote location: :keep do
+      %IPRange{
+        range: range,
+        first_ip: {first_a, first_b, first_c, first_d},
+        last_ip: {last_a, last_b, last_c, last_d}
+      } ->
+        case CIDR.parse_mask(range) do
+          {:ok, mask} -> <<12::int32, 2, mask, 0, 8, first_a, first_b, first_c, first_d, last_a, last_b, last_c, last_d>>
+          _ -> <<12::int32, 2, 255, 0, 8, first_a, first_b, first_c, first_d, last_a, last_b, last_c, last_d>>
+        end
+      %IPRange{
+        range: range,
+        first_ip: {first_a, first_b, first_c, first_d, first_e, first_f, first_g, first_h},
+        last_ip: {last_a, last_b, last_c, last_d, last_e, last_f, last_g, last_h}
+      } ->
+        <<36::int32, 3, 255, 0, 32,
+          first_a::size(16)-integer-unsigned, first_b::size(16)-integer-unsigned,
+          first_c::size(16)-integer-unsigned, first_d::size(16)-integer-unsigned,
+          first_e::size(16)-integer-unsigned, first_f::size(16)-integer-unsigned,
+          first_g::size(16)-integer-unsigned, first_h::size(16)-integer-unsigned,
+          last_a::size(16)-integer-unsigned, last_b::size(16)-integer-unsigned,
+          last_c::size(16)-integer-unsigned, last_d::size(16)-integer-unsigned,
+          last_e::size(16)-integer-unsigned, last_f::size(16)-integer-unsigned,
+          last_g::size(16)-integer-unsigned, last_h::size(16)-integer-unsigned>>
+    end
+  end
+
+  def decode(_) do
+    quote location: :keep do
+      <<8::int32, 2, bits, _, 4, first_a, first_b, first_c, first_d>> ->
```

Comment:
I removed my `_build` dir and recompiled with `mix compile --warnings-as-errors`. The project compiles on my machine running `Elixir 1.12.3 (compiled with Erlang/OTP 24)`.

Reply:
A regular compilation will not triggers those warning because the [type module](test/support/postgrex_types.ex) is not compiled there. If you run `MIX_ENV=test mix compile --warnings-as-errors` they should show up.